TestApp/src/main/java/com/tekchand/testapp/ui/main/SessionFrag.java

```diff
@@ -0,0 +1,193 @@
+        args.putString(ARG_PARAM2, param2);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            mParam1 = getArguments().getString(ARG_PARAM1);
+            mParam2 = getArguments().getString(ARG_PARAM2);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+
+        View root = inflater.inflate(R.layout.fragment_session, container, false);
+        return root;
+    }
+
+
+    TextView userName;
+    TextView email;
+    Button logOutBtn;
+
+
+    @Override
+    public void onViewCreated(@NonNull View root, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(root, savedInstanceState);
+        logOutBtn = root.findViewById(R.id.logoutbutton);
+        userName = root.findViewById(R.id.userNameTextView2);
+        email = root.findViewById(R.id.emailTextView2);
+    }
+
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        userName.setText("User Name: " + sharedPreferences.getString(NAME, null));
+        email.setText("Email Id: " + sharedPreferences.getString(EMAIL,null));
+
+        // When logout button pressed. It will go on main screen.
+        logOutBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                logoutUser();
+
+            }
+        });
+
+    }
+
+
+
+
+
+
+    /**
+     * Clear session details
+     * */
+    public void logoutUser(){
+       // SharedPreferences.Editor editor = getUserDetailsFromLogIn.getEditor();
+        // Clearing all data from Shared Preferences
+       // editor.clear();
+        //editor.commit();
+
+        // After logout redirect user to Loing Activity
+        Intent i = new Intent(getContext(), LogIn.class);
+        // Closing all the Activities
+        i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+
+        // Add new Flag to start new Activity
+        i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+        // Staring Login Activity
+        getContext().startActivity(i);
+    }
+
+    public interface GetUserDetailsFromLogIn {
+         ArrayList<String> getUserDetails();
+         SharedPreferences.Editor getEditor();
+    }
+
+
+
+    // TODO: Rename method, update argument and hook method into UI event
+    public void onButtonPressed(Uri uri) {
+        if (mListener != null) {
+            mListener.onFragmentInteraction(uri);
+        }
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof OnFragmentInteractionListener) {
```

Comment:
This is the correct way to attach fragments. Use it for other fragments as well.